### PR TITLE
[SPARK-30498][ML][PySpark] Fix some ml parity issues between python and scala

### PR DESCRIPTION
**Pull request by huaxingao**
### What changes were proposed in this pull request?
There are some parity issues between python and scala

### Why are the changes needed?
keep parity between python and scala

### Does this PR introduce any user-facing change?
Yes

### How was this patch tested?
existing tests

### CODE REVIEW

File: python/pyspark/ml/feature.py [hunk 1]
```diff
@@ -449,20 +449,6 @@ class BucketedRandomProjectionLSHModel(_LSHModel, _BucketedRandomProjectionLSHPa
...
-        """
-        return self._set(inputCol=value)
-
-    @since("3.0.0")
-    def setOutputCol(self, value):
-        """
-        Sets the value of :py:attr:`outputCol`.
-        """
-        return self._set(outputCol=value)
-
```
Comment (huaxingao) [inline python/pyspark/ml/feature.py]: These two setters are already in the super class ```_LSHModel ```

File: python/pyspark/ml/clustering.py [hunk 2]
```diff
@@ -1155,10 +1155,6 @@ def setSeed(self, value):
     def setTopicDistributionCol(self, value):
         """
         Sets the value of :py:attr:`topicDistributionCol`.
-
-        >>> algo = LDA().setTopicDistributionCol("topicDistributionCol")
-        >>> algo.getTopicDistributionCol()
-        'topicDistributionCol'
```
Comment (huaxingao) [inline python/pyspark/ml/clustering.py]: I will remove this. Normally there is no doctest in the setters.

File: python/pyspark/ml/feature.py [hunk 3]
```diff
@@ -873,20 +859,6 @@ def setOutputCol(self, value):
         """
         return self._set(outputCol=value)
 
-    @since("3.0.0")
```
Comment (zhengruifeng) [inline python/pyspark/ml/feature.py]: setMinTF and setBinary exist in the scala side

Comment (zhengruifeng) [reply python/pyspark/ml/feature.py]: Oh， they were defined twice

**Review by zhengruifeng — APPROVED** [review]

**Comment by zhengruifeng** [comment]
Merged to master!

**Comment by huaxingao** [comment]
Thank you very much!!

**Pull request closed without merging**